server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -223,6 +226,10 @@ public class IndexShard extends AbstractIndexShardComponent implements IndicesCl
     private final ReplicationTracker replicationTracker;
     private final IndexStorePlugin.SnapshotCommitSupplier snapshotCommitSupplier;
     private final Engine.IndexCommitListener indexCommitListener;
+    private final Map<String, Boolean> fieldHasValue;
```

Comment:
we were discussing with @original-brownbear this map and whether we can avoid creating it, and reuse the data structures that are already hot, hence perhaps keep track of the FieldInfos directly. This would allow us to change the scope of the MappedFieldType method to something more targeted to the information we need to obtain e.g. ft.fieldHasValue(FieldInfos) . This feels like a better design than introducing `concreteFieldName` that has weird semantics and broad meaning.

Reply:
concreteFieldName didn't resonate with me from the start, this solution instead makes much more sense and would avoid major changes that introducing concreteFieldName would have.
So, sounds like a plan. Thanks for the feedback!